### PR TITLE
CPU Support for Data_stats / Dropna / DifferenceLag / Hash_bucket / Hashed_cross

### PR DESCRIPTION
**Pull request by albert17**
Related to #534

Adds support for cpu-backed data in `data_stats` (and test coverage).
Adds support for cpu-backed data in `dropna` (and test coverage).
Adds support for cpu-backed data in `drifference_lag` (and test coverage).
Adds support for cpu-backed data in `Hash_bucket` (and test coverage).
Adds support for cpu-backed data in `Hash_crossed` (and test coverage).

### CODE REVIEW

File: nvtabular/ops/difference_lag.py [hunk 1]
```diff
@@ -58,19 +59,19 @@ def __init__(self, partition_cols, shift=1):
...
+            mask = df[self.partition_cols] == df[self.partition_cols].shift(shift)
+            if _is_dataframe_object(mask):
                 mask = mask.fillna(False).all(axis=1)
             mask[mask == False] = None  # noqa pylint: disable=singleton-comparison
 
             for col in columns:
-                output[self._column_name(col, shift)] = (gdf[col] - gdf[col].shift(shift)) * mask
-        return cudf.DataFrame(output)
+                output[self._column_name(col, shift)] = (df[col] - df[col].shift(shift)) * mask
+        return _create_frame(output, df)
```
Comment (rjzamora) [inline nvtabular/ops/difference_lag.py:74]: Other ops have typically done something like: `type(df)(output)`

Comment (albert17) [reply nvtabular/ops/difference_lag.py:74]: @rjzamora Oh, just casting will work?

File: nvtabular/ops/hashed_cross.py [hunk 2]
```diff
@@ -53,19 +54,19 @@ def __init__(self, num_buckets: Union[int, Dict[str, int]]):
         self.num_buckets = num_buckets
 
     @annotate("HashedCross_op", color="darkgreen", domain="nvt_python")
-    def transform(self, columns: ColumnNames, gdf: cudf.DataFrame) -> cudf.DataFrame:
-        new_gdf = cudf.DataFrame()
+    def transform(self, columns: ColumnNames, df: DataFrameType) -> DataFrameType:
+        new_df = type(df)({})
```
Comment (rjzamora) [inline nvtabular/ops/hashed_cross.py:58]:
```suggestion
        new_df = type(df)()
```
I don't think you need the empty dict here.

**Review by rjzamora — APPROVED** [review]
I left a minor suggestion, but looks good - Thanks @albert17 !

**Review by benfred — APPROVED** [review]

**Pull request merged**